common/transport/amqp/src/amqp_common_errors.ts

```diff
@@ -102,6 +102,13 @@ export function translateError(message: string, amqpError: Error): AmqpTransport
     }
   } else if (amqpError instanceof Amqp10Errors.AuthenticationError) {
     error = new errors.UnauthorizedError(message);
+  } else if ((<any>amqpError).code) {
+    switch ((<any>amqpError).code) {
+      case 'ENOENT':
+      case 'ENOTFOUND':
+      default:
```

Comment:
OK.  Why have a switch at all?

Reply:
I wanted to look at the list of errors and discriminate but looks like this will have to wait ([https://nodejs.org/api/errors.html#errors_common_system_errors](https://nodejs.org/api/errors.html#errors_common_system_errors))

---
In reply to: [139542426](https://github.com/Azure/azure-iot-sdk-node/pull/83#discussion_r139542426) [](ancestors = 139542426)